### PR TITLE
feat: Added MissingDataError + Data processing tutorial

### PR DESCRIPTION
**Pull request by Marsmaennchen221**
Related to Issue #380 

### Summary of Changes

Added a new data processing tutorial

`from_rows` and `from_columns` now raise a `MissingDataError` if the provided lists of rows/columns are empty
if none of the rows of a table in `filter_rows` match with the query the function now returns an empty table with the same schema as the original table

### CODE REVIEW

**Comment by Marsmaennchen221** [comment]
I'm not sure if "docs:" in the title is the right type since I also added an Exception Class and altered three methods in the Table class

File: Runtime/safe-ds/safeds/data/tabular/_table.py [hunk 1]
```diff
@@ -163,9 +164,14 @@ def from_rows(rows: list[Row]) -> Table:
 
         Raises
         ------
+        MissingDataError
+            If an empty list without a row is given.
```
Comment (WinPlay02) [inline Runtime/safe-ds/safeds/data/tabular/_table.py:168]:
```suggestion
            If an empty list is given.
```

This comment seems a bit redundant

File: Runtime/safe-ds/safeds/data/tabular/_table.py [hunk 2]
```diff
@@ -195,9 +201,14 @@ def from_columns(columns: list[Column]) -> Table:
 
         Raises
         ------
+        MissingDataError
+            If an empty list without a column is given.
```
Comment (WinPlay02) [inline Runtime/safe-ds/safeds/data/tabular/_table.py:205]:
```suggestion
            If an empty list is given.
```

File: Runtime/safe-ds/safeds/exceptions/_data_exceptions.py [hunk 3]
```diff
@@ -94,3 +94,12 @@ class ColumnLengthMismatchError(Exception):
 
     def __init__(self, column_info: str):
         super().__init__(f"The length of at least one column differs: \n{column_info}")
+
+
+class MissingDataError(Exception):
+    """
+    Exception raised when a function is given not enough data to succeed.
```
Comment (WinPlay02) [inline Runtime/safe-ds/safeds/exceptions/_data_exceptions.py:101]:
```suggestion
    Exception raised if a function is not given enough data to succeed.
```

File: docs/Stdlib/python/Tutorials/data_processing.md [hunk 4]
```diff
@@ -1,78 +1,314 @@
...
 
-    ```python
-    # provide the name of the column
-    column = table.get_column("TheColumnName")
-    ```
+table_from_json = Table.from_json("path/to/your/json/file.json")
+```
 
-1. Combine a list of rows to a table
+This table has the following output:
```
Comment (WinPlay02) [inline docs/Stdlib/python/Tutorials/data_processing.md:22]:
```suggestion
This table produces the following output:
```

File: docs/Stdlib/python/Tutorials/data_processing.md [hunk 5]
```diff
@@ -1,78 +1,314 @@
...
+| 5       | 3       | 1       |
 
-1. Combine a list of columns to a table
+## Create Rows and Columns
 
-    ```python
-    # make sure the columns have the same amount of rows
-    table = Table.from_columns([column1, column2])
-    ```
+Columns and Rows can be created with python Iterables or extracted from an existing Table.
```
Comment (WinPlay02) [inline docs/Stdlib/python/Tutorials/data_processing.md:34]:
```suggestion
Columns and rows can be created with python Iterables or extracted from an existing table.
```

File: docs/Stdlib/python/Tutorials/data_processing.md [hunk 6]
```diff
@@ -1,78 +1,314 @@
...
 
-1. Drop columns from a table or keep only specified columns in a table
+If we want to create a Row we have to provide a [TableSchema][safeds.data.tabular.typing.TableSchema]. This TableSchema contains information about the Columns, which data their storing and which order they have.
```
Comment (WinPlay02) [inline docs/Stdlib/python/Tutorials/data_processing.md:36]:
```suggestion
If we want to create a Row we have to provide a [TableSchema][safeds.data.tabular.typing.TableSchema]. This TableSchema contains information about the columns, which data they are storing and which order they have.
```

File: docs/Stdlib/python/Tutorials/data_processing.md [hunk 7]
```diff
@@ -1,78 +1,314 @@
...
+The TableSchema takes a dictionary as input with the names of the Columns as keys and their [ColumnTypes][safeds.data.tabular.typing.ColumnType] as values.
```
Comment (WinPlay02) [inline docs/Stdlib/python/Tutorials/data_processing.md:37]:
```suggestion
The TableSchema takes a dictionary as input with the names of the columns as keys and their [ColumnTypes][safeds.data.tabular.typing.ColumnType] as values.
```

File: docs/Stdlib/python/Tutorials/data_processing.md [hunk 8]
```diff
@@ -1,78 +1,314 @@
...
 
-    ```python
-    # drop specified columns
-    table = table.drop_columns(["NameOfColumn1", "NameOfColumn2"])
+A Table has a TableSchema as well which is created when creating a Table.
```
Comment (WinPlay02) [inline docs/Stdlib/python/Tutorials/data_processing.md:39]:
```suggestion
A Table contains a TableSchema as well, which is created when creating a Table.
```

File: docs/Stdlib/python/Tutorials/data_processing.md [hunk 9]
```diff
@@ -1,78 +1,314 @@
...
+
+Sometimes it might be useful to remove some columns. Therefore, you can create a new table which will either [keep][safeds.data.tabular.Table.keep_columns] or [drop][safeds.data.tabular.Table.drop_columns] some columns from the original table.
+
+```python
+table_without_column_1 = table.drop_columns(["column1"])
+
+table_with_only_column_1_and_2 = table.keep_columns(["column1", "column2"])
+```
+
+First the table dropped column1. This leaves a new Table with column2 and column3:
```
Comment (WinPlay02) [inline docs/Stdlib/python/Tutorials/data_processing.md:116]:
```suggestion
First the column1 is dropped. This leaves a new table containing column2 and column3:
```

File: docs/Stdlib/python/Tutorials/data_processing.md [hunk 10]
```diff
@@ -1,78 +1,314 @@
...
+
+| column2 | column3 |
+| ------- | ------- |
+| 2       | 1       |
+| 2       | 1       |
+| 1       | 1       |
+| 6       | 1       |
+| 3       | 1       |
+
+In the second example a new Table is created keeping column1 and column2:
```
Comment (WinPlay02) [inline docs/Stdlib/python/Tutorials/data_processing.md:126]:
```suggestion
In the second example a new table is created by keeping only column1 and column2:
```

File: docs/Stdlib/python/Tutorials/data_processing.md [hunk 11]
```diff
@@ -1,78 +1,314 @@
...
+| ------- | ------- |
+| 3       | 2       |
+| 4       | 2       |
+| 8       | 1       |
+| 6       | 6       |
+| 5       | 3       |
+
+## Imputing missing values
+
+If your data has missing values, it can provide a challenge for some machine learning models.
```
Comment (WinPlay02) [inline docs/Stdlib/python/Tutorials/data_processing.md:138]:
```suggestion
If your data is missing values, it might be challanging for some machine learning models.
```

File: docs/Stdlib/python/Tutorials/data_processing.md [hunk 12]
```diff
@@ -1,78 +1,314 @@
...
+For this you can use the [Imputer][safeds.data.tabular.transformation.Imputer] to deal with these values.
```
Comment (WinPlay02) [inline docs/Stdlib/python/Tutorials/data_processing.md:139]:
```suggestion
The [Imputer][safeds.data.tabular.transformation.Imputer] can be used to deal with these values.
```

File: docs/Stdlib/python/Tutorials/data_processing.md [hunk 13]
```diff
@@ -1,78 +1,314 @@
...
+
+table_with_nan = Table({"column1": [None, 4, 8, None, 5],
+                        "column2": [2, 2, 1, 6, 3],
+                        "column3": [1, 1, 1, 1, 1]})
+
+imputer = Imputer(Imputer.Strategy.Median())
+table2_imp = imputer.fit_transform(table_with_nan, ["column1"])
+```
+
+The following Table has its `nan` values replaced with the median of column1:
```
Comment (WinPlay02) [inline docs/Stdlib/python/Tutorials/data_processing.md:153]:
```suggestion
The `nan` values in the following table are replaced with the median value of column1:
```

File: docs/Stdlib/python/Tutorials/data_processing.md [hunk 14]
```diff
@@ -1,78 +1,314 @@
...
+| ------- | ------- | ------- |
+| 5.0     | 2.0     | 1.0     |
+| 4.0     | 2.0     | 1.0     |
+| 8.0     | 1.0     | 1.0     |
+| 5.0     | 6.0     | 1.0     |
+| 5.0     | 3.0     | 1.0     |
+
+## Dealing with strings as data
+
+For the following three examples we will use a new Table with strings:
```
Comment (WinPlay02) [inline docs/Stdlib/python/Tutorials/data_processing.md:165]:
```suggestion
In the following three examples we will use a new Table with strings:
```

File: docs/Stdlib/python/Tutorials/data_processing.md [hunk 15]
```diff
@@ -1,78 +1,314 @@
...
+| ------------- | ---------------- | --------------- | ------------- | -------------- | -------------- |
+| 0.0           | 0.0              | 1.0             | 0.0           | 1.0            | 0.0            |
+| 0.0           | 0.0              | 0.0             | 1.0           | 1.0            | 0.0            |
+| 1.0           | 0.0              | 0.0             | 0.0           | 0.0            | 1.0            |
+| 0.0           | 0.0              | 0.0             | 1.0           | 0.0            | 1.0            |
+| 0.0           | 1.0              | 0.0             | 0.0           | 1.0            | 0.0            |
+
+### Using an OrdinalEncoder to deal with strings in a column
+
+The third method to deal with strings in your data is the [OrdinalEncoder][safeds.data.tabular.transformation.OrdinalEncoder]. This encoder works like the label encoder but keeps an order of the encoded values. This order has to be provided with the fitting method.
```
Comment (WinPlay02) [inline docs/Stdlib/python/Tutorials/data_processing.md:220]:
```suggestion
The third method to deal with strings in your data is the [OrdinalEncoder][safeds.data.tabular.transformation.OrdinalEncoder]. This encoder works like the label encoder but keeps the order of the encoded values. This order has to be provided when creating the encoder.
```

File: docs/Stdlib/python/Tutorials/data_processing.md [hunk 16]
```diff
@@ -1,78 +1,314 @@
...
+| 2       | 1       |
+| 11      | 0       |
+
+## Statistics of a Column
+
+To drop or keep certain columns it might be useful to evaluate which columns are important enough to keep and which columns can be dropped.
+For this you can use the statistics of a Column.
+The [ColumnStatistics][safeds.data.tabular.ColumnStatistics] class provides multiple functions like [`min`][safeds.data.tabular.ColumnStatistics.min], [`max`][safeds.data.tabular.ColumnStatistics.max], [`mode`][safeds.data.tabular.ColumnStatistics.mode], [`mean`][safeds.data.tabular.ColumnStatistics.mean], [`median`][safeds.data.tabular.ColumnStatistics.median] and others.
+
+If you want a summary over all the columns you can use the [`summary`][safeds.data.tabular.Table.summary] function which provides a table with all the statistics for each column.
```
Comment (WinPlay02) [inline docs/Stdlib/python/Tutorials/data_processing.md:246]:
```suggestion
If you want a summary over all the columns you can use the [`summary`][safeds.data.tabular.Table.summary] function which provides a table containing the statistics for each column.
```

File: docs/Stdlib/python/Tutorials/data_processing.md [hunk 17]
```diff
@@ -1,78 +1,314 @@
...
+This outputs the following Table:
+
+| column1 | column2 | column3 |
+| ------- | ------- | ------- |
+| 3       | 2       | 1       |
+| 8       | 1       | 1       |
+
+## Write a Table into a csv or json file
+
+If you finished your work with the data you might want to store the data in a file.
```
Comment (WinPlay02) [inline docs/Stdlib/python/Tutorials/data_processing.md:307]:
```suggestion
If you finished your work with the data you might want to store the result in a file.
```

**Review by WinPlay02 — CHANGES_REQUESTED** [review]
I'm a bit unsure about the capitalization of Row, Table, Column, etc.
I suggested some changes, but left the capitalization unchanged as it can be useful in most of the cases.

File: docs/Stdlib/python/Tutorials/data_processing.md (hunk 5 above)
Comment (Marsmaennchen221) [reply docs/Stdlib/python/Tutorials/data_processing.md:34]: I think the capitalization might be useful because the tutorial is referencing the Row and Table Classes

File: docs/Stdlib/python/Tutorials/data_processing.md (hunk 9 above)
Comment (Marsmaennchen221) [reply docs/Stdlib/python/Tutorials/data_processing.md:116]:
```suggestion
First the column1 is dropped. This leaves a new Table containing column2 and column3:
```
A new instance of the Table class is created, therefore i would keep the capitalization

File: docs/Stdlib/python/Tutorials/data_processing.md (hunk 10 above)
Comment (Marsmaennchen221) [reply docs/Stdlib/python/Tutorials/data_processing.md:126]:
```suggestion
In the second example a new Table is created by keeping only column1 and column2:
```
A new instance of the Table class is created, therefore i would keep the capitalization

**Comment by WinPlay02** [comment]
Alright. In that case your suggestions are fine.

File: docs/Stdlib/python/Tutorials/data_processing.md (hunk 9 above)
Comment (WinPlay02) [reply docs/Stdlib/python/Tutorials/data_processing.md:116]:
```suggestion
First column1 is dropped. This leaves a new Table containing column2 and column3:
```

**Review by WinPlay02 — CHANGES_REQUESTED** [review]
There is still one conversation remaining

**Review by WinPlay02 — APPROVED** [review]
